src/libraries/System.Reflection.Emit/tests/PersistableAssemblyBuilder/AssemblySaveConstructorBuilderTests.cs

```diff
@@ -0,0 +1,188 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Xunit;
+
+namespace System.Reflection.Emit.Tests
+{
+    [ConditionalClass(typeof(PlatformDetection), nameof(PlatformDetection.IsNotBrowser))]
+    public class AssemblySaveConstructorBuilderTests
+    {
+        [Fact]
+        public void DefineConstructorsTest()
+        {
+            AssemblyBuilder ab = AssemblySaveTools.PopulateAssemblyBuilderTypeBuilderAndSaveMethod(out TypeBuilder type, out MethodInfo _);
+            ConstructorBuilder constructor = type.DefineDefaultConstructor(MethodAttributes.Public);
+            ConstructorBuilder constructor2 = type.DefineConstructor(MethodAttributes.Public, CallingConventions.Standard, new[] { typeof(int) });
+            ILGenerator il = constructor2.GetILGenerator();
+            il.Emit(OpCodes.Ret);
+            type.CreateType();
+
+            ConstructorInfo[] ctors = type.GetConstructors();
+            Assert.Equal(2, ctors.Length);
+
+            Assert.Equal(constructor, type.GetConstructor(Type.EmptyTypes));
+            Assert.Equal(ctors[0], type.GetConstructor(Type.EmptyTypes));
+            Assert.Equal(ctors[1], type.GetConstructor(new[] { typeof(int) }));
+            Assert.Null(type.GetConstructor(new[] { typeof(string) }));
+        }
+
+        [Fact]
+        public void DefineDefaultConstructor_WithTypeBuilderParent()
+        {
+            AssemblyBuilder ab = AssemblySaveTools.PopulateAssemblyBuilderTypeBuilderAndSaveMethod(out TypeBuilder type, out MethodInfo _);
+            type.CreateType();
+            TypeBuilder child = ab.GetDynamicModule("MyModule").DefineType("ChildType", TypeAttributes.Public | TypeAttributes.Class);
+            child.SetParent(type);
+            child.DefineDefaultConstructor(MethodAttributes.Family);
+            child.CreateType();
+
+            ConstructorInfo[] ctors = child.GetConstructors(BindingFlags.NonPublic | BindingFlags.Instance);
+            Assert.Equal(1, ctors.Length);
+            Assert.True(ctors[0].IsFamily);
+            Assert.Empty(child.GetConstructors(BindingFlags.Public | BindingFlags.Instance));
+        }
+
+
+        [Fact]
+        public void DefineDefaultConstructor_GenericParentCreated_Works()
+        {
+            AssemblyBuilder ab = AssemblySaveTools.PopulateAssemblyBuilderTypeBuilderAndSaveMethod(out TypeBuilder type, out MethodInfo _);
+            type.DefineGenericParameters("T");
```

Comment:
Is there a way to get the "T" back and verify? e.g. `GetGenericMethodDefinition().GetGenericArguments()[0].Name`

Reply:
Added `Assert.Equal("T", type.GetGenericTypeDefinition().GetGenericArguments()[0].Name);` as this is generic type definition not a method